### ISSUE TITLE
Debugger add configuration not showing C++

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode-cpptools#1287]
_From @rajinder-yadav on November 25, 2017 8:45_

- VSCode Version: Code - Insiders 1.19.0-insider (89b158e11cb1c3fe94a3876222478ed2d0549fc8, 2017-11-24T06:07:59.584Z)
- OS Version: Darwin x64 17.2.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
vscode-markdownlint|Dav|0.12.0
EditorConfig|Edi|0.11.1
vscode-eslint|dba|1.4.3
tslint|eg2|1.0.23
cpptools|ms-|0.14.2
debugger-for-chrome|msj|3.5.0
autoimport|ste|1.5.3
vim|vsc|0.10.5
markdown-pdf|yza|0.1.7

I am not shown C++ when I try to create a Debugger configuration for a new project. I have to click on install plugin, then click on the already install C/C++ plugin. Go back to Debugger and select, Create configuration for the C++ option to show up.

---

Steps to Reproduce:

1. Click on Debugger icon
2. Select "Add Configuration..."

Reproduces without extensions: Yes/No

_Copied from original issue: Microsoft/vscode#39124_

**Comment by bobbrow** [comment]
Does this happen every time you create a new project, or just the first time you go to the debugger after installing an update?  This may be a side effect of a change we made that removed the second "reload window" prompt.

**Comment by sean-mcmanus** [comment]
@rajinder-yadav I'm not able to repro the issue -- I get a dropdown when the launch.json is opened that explains that a reload is required.

**Comment by pieandcakes** [comment]
@rajinder-yadav also, make sure you have a *.cpp or *.c file open so our extension activates. If our extension is not activated then you won't get those options. This happens if VS Code doesn't know you have a *.cpp or *.c style file open.

**Comment by sean-mcmanus** [comment]
@pieandcakes Having a .cpp/.c file open should not matter. We activate on "*" before the reload is done. I tested with just a folder with no .cpp/.c files and I see the dropdown appear explaining that reload is required for the snippets will be available (and doing at reload window makes the snippets appear).

**Comment by rajinder-yadav** [comment]
I can create this issue every time, first time I don't see C++, if I select again I see C++ show up.

I am using MacOS and have C++ (gcc and gdb) installed using homebrew if that helps with investigation.

Here is a video showing  the issue:

https://www.youtube.com/watch?v=l3vULqTn2Lw&feature=youtu.be

**Comment by pieandcakes** [comment]
@sean-mcmanus that is not true. We activate on "*" ONLY for the first activation. When we rewrite the `package.json` we replace it with these activation events:
```
  "activationEvents": [
    "onLanguage:cpp",
    "onLanguage:c",
    "onCommand:extension.pickNativeProcess",
    "onCommand:extension.pickRemoteNativeProcess",
    "onCommand:C_Cpp.ConfigurationEdit",
    "onCommand:C_Cpp.ConfigurationSelect",
    "onCommand:C_Cpp.SwitchHeaderSource",
    "onCommand:C_Cpp.Navigate",
    "onCommand:C_Cpp.GoToDeclaration",
    "onCommand:C_Cpp.PeekDeclaration",
    "onCommand:C_Cpp.ToggleErrorSquiggles",
    "onCommand:C_Cpp.ToggleIncludeFallback",
    "onCommand:C_Cpp.ShowReleaseNotes",
    "onCommand:C_Cpp.ResetDatabase",
    "onCommand:C_Cpp.PauseParsing",
    "onCommand:C_Cpp.ResumeParsing",
    "onCommand:C_Cpp.ShowParsingCommands",
    "onCommand:C_Cpp.TakeSurvey",
    "onDebug"
  ],
```

**Comment by pieandcakes** [comment]
@rajinder-yadav I noticed you are using VSCode Insiders. does this also happen with a released version of VSCode stable? I don't have a Mac with me at the moment but I can give it a try tomorrow with the latest insiders. As Sean mentioned above, I haven't seen this happen with VSCode stable.

**Comment by rajinder-yadav** [comment]
@pieandcakes yes so I switched to using code-insiders hoping it was fixed.

**Comment by sean-mcmanus** [comment]
Ah, thanks for the video. Yeah, I'm able to repro that. It's what @pieandcakes was talking about -- our extension activate is called for the 1st time when the Add Configuration is used, but our extension isn't finished loading so the list isn't complete until the 2nd time it's pressed. We may be able to fix this (not sure yet). It repros on Insiders and non-Insiders.

**Comment by rajinder-yadav** [comment]
@sean-mcmanus glad you were able to repro and have an idea of cause, hope it's fixable.

**Comment by sean-mcmanus** [comment]
@rajinder-yadav Yeah, the fix worked, thanks for reporting this. It should be fixed in our next 0.14.4 update. Having a C/C++ file open beforehand would be one workaround. You could also manually implement the fix yourself via opening `/Users/<user>/.vscode/extensions/ms-vscode.cpptools-0.14.3/out/src/main.js`, commenting out lines 79 and 80, and moving line 119 (`DebuggerExtension.activate();`) to be before line 95 (`processRuntimeDependencies(() => {`).

**Comment by rajinder-yadav** [comment]
Thank you guys!

**Comment by bobbrow** [comment]
This has been addressed in version 0.14.4
